src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -27,3 +29,108 @@ def show_version(cmd):  # pylint: disable=unused-argument
     from azure.cli.core.util import get_az_version_json
     versions = get_az_version_json()
     return versions
+
+
+def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many-locals, too-many-statements, too-many-branches, no-member, unused-argument
+    import os
+    import platform
+    import sys
+    import subprocess
+    import azure.cli.core.telemetry as telemetry
+    from azure.cli.core import __version__ as local_version
+    from azure.cli.core._environment import _ENV_AZ_INSTALLER
+    from azure.cli.core.extension import get_extensions, WheelExtension
+    from distutils.version import LooseVersion
+    from knack.util import CLIError
+
+    update_cli = True
+    from azure.cli.core.util import get_latest_from_github
+    try:
+        latest_version = get_latest_from_github()
+        if latest_version and LooseVersion(latest_version) <= LooseVersion(local_version):
+            logger.warning("You already have the latest azure-cli version: %s", local_version)
+            update_cli = False
+            if not update_all:
+                return
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.debug("Failed to get the latest version. %s", str(ex))
+    exts = [ext.name for ext in get_extensions(ext_type=WheelExtension)] if update_all else []
+
+    exit_code = 0
+    if update_cli:
+        logger.warning("Your current Azure CLI version is %s. Latest version available is %s.",
+                       local_version, latest_version)
+        from knack.prompting import prompt_y_n
+        if not yes:
+            confirmation = prompt_y_n("Please check the release notes first: https://docs.microsoft.com/"
+                                      "cli/azure/release-notes-azure-cli\nDo you want to continue?", default='y')
+            if not confirmation:
+                telemetry.set_success("Upgrade stopped by user")
+                return
+        installer = os.getenv(_ENV_AZ_INSTALLER)
+        if installer == 'DEB':
+            from azure.cli.core.util import in_cloud_console
+            if in_cloud_console():
+                raise CLIError("az upgrade is not supported in Cloud Shell.")
+            apt_update_cmd = 'apt-get update'.split()
+            az_update_cmd = 'apt-get install --only-upgrade -y azure-cli'.split()
+            if os.geteuid() != 0:  # pylint: disable=no-member
+                apt_update_cmd.insert(0, 'sudo')
+                az_update_cmd.insert(0, 'sudo')
+            subprocess.call(apt_update_cmd)
+            logger.debug("Update azure cli with '%s'", " ".join(apt_update_cmd))
+            exit_code = subprocess.call(az_update_cmd)
```

Comment:
What if previous cmd such as `apt-get update`, `apt-get install --only-upgrade ...` fail ? It seems that `exit_code` should accept all `subprocess`'s return code.

Reply:
Good catch. Added check on the exit code of the first command.